File: frontend/angular-secretroom-website/src/app/model/order.ts
  import {Product} from "./product";
  import {ShippingOption} from "./shipping-option";
  import {Customer} from "./customer";

  export class Order {

    constructor(
      public id: number | null,
      public placementDate: Date,
      public shippingOption?: ShippingOption,
      public shippingAddress?: ShippingAddress, // добавьте это свойство
      public customer? : Customer,
      public delivery?: number,
      public totalQuantity?: number,
      public totalAmount?: number,
      public totalAmountOrder?: number,
      public items?: OrderItem[],
      public comment?: string

  )
    {
      this.id = id;
      this.placementDate = placementDate;
      this.shippingOption = shippingOption;
      this.shippingAddress = shippingAddress; // и это
      this.customer = customer;
      this.delivery = delivery;
      this.totalQuantity = totalQuantity;
      this.totalAmount = totalAmount;
      this.totalAmountOrder = totalAmountOrder;
      this.items = items;
      this.comment = comment;

    }
  }

  export interface OrderItem {
    product: Product;
    sizeType?: string;
    amount: number;
    quantity: number;
  }


  export class ShippingAddress {
    constructor(
      public street?: string,
      public city?: string,
      public country?: string,
      public zipCode?: string,
      public phoneNumber?: string,
      public email?: string
    ) {
      this.street = street;
      this.city = city;
      this.country = country;
      this.zipCode = zipCode;
    }
  }

File: backend/src/main/java/com/secretroomwebsite/order/Order.java
package com.secretroomwebsite.order;

import com.secretroomwebsite.adress.Address;
import com.secretroomwebsite.customer.Customer;
import com.secretroomwebsite.shipping.Shipping;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Pattern;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.util.Date;
import java.util.List;

@Entity
@Table(name = "orders")
@Getter
@Setter
@AllArgsConstructor
@ToString

public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name="order_tracking_number")
    private String orderTrackingNumber;

    @NotNull
    @Enumerated(EnumType.STRING)
    private OrderStatus status;


    @CreationTimestamp
    private Date placementDate;

    @ManyToOne
    @JoinColumn(name = "shipping_option_id")
    @NotNull
    private Shipping shippingOption;

    @NotNull
    private Integer totalQuantity;

    @NotNull
    private Double totalAmount;

    @NotNull
    private Double shippingCost;

    @NotNull
    private Double totalAmountOrder;

    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL)
    @NotEmpty
    @ToString.Exclude
    private List<OrderItem> items;

    @ManyToOne
    @JoinColumn(name = "customer_id")
    private Customer customer;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "shipping_address_id", referencedColumnName = "id")
    private Address shippingAddress;

    String comment;

    public Order() {}


}
